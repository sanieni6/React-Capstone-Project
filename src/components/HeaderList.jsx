/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';

const HeaderList = ({ year, handleClick }) => (
  <li key={year}>

    <button
      onClick={() => handleClick(year)}
      onKeyDown={() => handleClick(year)}
      key={year}
      type="button"
      className="tags-btn"
    >
      {year}
    </button>

  </li>
);

HeaderList.propTypes = {
  year: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default HeaderList;
