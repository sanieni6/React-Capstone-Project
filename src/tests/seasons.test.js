import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import HeaderList from '../components/HeaderList';

describe('HeaderList component', () => {
  test('renders correctly', () => {
    const handleClick = jest.fn();
    const { container } = render(
      <Router>
        <HeaderList year={2021} handleClick={handleClick} />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });

  test('clicking a year button calls the handleClick function', () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Router>
        <HeaderList year={2021} handleClick={handleClick} />
      </Router>,
    );
    const button = getByText('2021');
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(handleClick).toHaveBeenCalledWith(2021);
  });

  test('displays the correct year', () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Router>
        <HeaderList year={2021} handleClick={handleClick} />
      </Router>,
    );
    expect(getByText('2021')).toBeInTheDocument();
  });
});
