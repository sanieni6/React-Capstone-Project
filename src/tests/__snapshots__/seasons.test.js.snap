// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderList component renders correctly 1`] = `
<div>
  <li>
    <button
      class="tags-btn"
      type="button"
    >
      2021
    </button>
  </li>
</div>
`;
